astro/src/layouts/EscapeRoomContentLayout.astro

```diff
@@ -0,0 +1,271 @@
+    <div class="container hero text-white">
+        <h1 class="visually-hidden">Accessible Escape Room Digital Content</h1>
+        <div class="display-2 py-4 mx-auto text-center">
+            <span class="text-nowrap">{title}</span>
+        </div>
+    </div>
+    <nav
+        class="tab-nav-container overflow-hidden w-100"
+        style="position:sticky; top:var(--er-header-offset,0); z-index:1020;"
+    >
+      <div class="navbar w-100 px-0">
+        <ul class="navbar-nav flex-row flex-wrap w-100 justify-content-md-center justify-content-start gap-2 gap-md-4">
+          <li class="nav-item ms-md-5">
+            <a href="#instructions" class="nav-link px-2 text-nowrap">Instructions</a>
+          </li>
+          <li class="nav-item">
+            <a href="#elements" class="nav-link px-2 text-nowrap">Elements</a>
+          </li>
+          <li class="nav-item">
+            <a href="#layout" class="nav-link px-2 text-nowrap">Room Layout</a>
+          </li>
+          <li class="nav-item">
+            <a href="#leaderboard" class="nav-link px-2 text-nowrap">Leaderboard</a>
+          </li>
+        </ul>
+      </div>
+      <style>
+        .tab-nav-container .nav-link.active {
+          border-bottom: 3px solid #000 !important;
+        }
+      </style>
+    </nav>
+
+    <main id="main-content" role="main" class="bg-white text-black">
+
+
+    <div class="container py-5 color-black">
+      <section id="instructions" class="tab-content-section py-5 mb-5">
+        <h2 class="text-center">Instructions</h2>
+        <slot name="instructions" />
+      </section>
+
+      <section id="elements" class="tab-content-section py-5 mb-5">
+        <h2 class="text-center">Elements</h2>
+        <slot name="elements" />
+      </section>
+
+      <section id="layout" class="tab-content-section py-5 mb-5">
+        <h2 class="text-center">Room Layout</h2>
+        <slot name="layout" />
+      </section>
+
+      <section id="leaderboard" class="tab-content-section py-5 mb-4">
+        <h2 class="text-center">Leaderboard & Results</h2>
+        <div class="table-responsive">
+        <table class="table table-striped">
+          <thead>
+            <tr>
+            <th>Rank</th>
+            <th>Team Name</th>
+            <th>Completion Time</th>
+            </tr>
+          </thead>
+          <tbody id="er-leaderboard-body">
+            <tr>
+            <td colspan="3" class="text-muted">Leaderboard will be populated programmatically.</td>
+            </tr>
+          </tbody>
+        </table>
+        </div>
+      </section>
+
+      <div id="er-content-staging" hidden>
+        <slot />
+      </div>
+    </div>
+    </main>
+
+    <script>
+      // @ts-nocheck
```

Comment:
Thoughts on this code being it's own js file? Maybe some code comments too for what it's calculating just so it's quicker to read?